org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -496,6 +519,28 @@ public void run() throws EventException {
 		}
 	}
 	
+	private void checkStartPaused() throws EventException {
+		
+		if (!isPauseOnStart()) { 
+			return;
+		}
+		
+		List<U> items = getSubmissionQueue();
+		if (items!=null && items.size()>0) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1155)